### ISSUE TITLE
chia:latest is breaking 0.7.2's "summary" screen

### ISSUE THREAD

**Issue by joshpainter** [guydavis/machinaris#682]
Lots of recent updates to chia:latest repo (looks like 1.4 drop imminent) but this is breaking the 0.7.2 release in weird ways (guessing cause CLI has changed a bit?). Seems like 0.7.2 should be asking for chia:v1.3.5 instead of chia:latest to fix this. I reverted back to 0.7.1 in my docker.compose file as workaround for now.

```
version: '3.7'
services:
    machinaris:
        image: ghcr.io/guydavis/machinaris:v0.7.1
```

**Comment by guydavis** [comment]
Hi, thanks for the report.  I could be doing something wrong here. Let's check.

Were you running the `machinaris:latest` image [v0.7.2](https://github.com/guydavis/machinaris/blob/main/scripts/forks/chia_install.sh) that was built [3 weeks ago](https://github.com/guydavis/machinaris/actions/workflows/main-chia.yaml).  This was linked to [chia latest branch](https://github.com/Chia-Network/chia-blockchain/tree/latest) (now 23 days old).

All [three Machinaris streams](https://github.com/guydavis/machinaris/wiki/Releases) should be using that Chia latest branch as shown here: 

![image](https://user-images.githubusercontent.com/12936/171959372-87b9a379-7efa-437c-9cac-4292bc182533.png)

So to sum up, the Chia branch *should* be pulled on the Github build servers when the Docker image is built, not when you run the container.  However, I definitely would appreciate a pointer if I'm doing something wrong here.

**Comment by joshpainter** [comment]
I'm not an expert on Docker/Github/GH build servers either, but I kind of assumed that just the "chia-blockchain:latest" reference is stored in the image and it is dynamically downloaded (images are just layers on top of other layers, all text files that resolve repos, right?) at the time that the image is actually built on the docker host. So for example: in your case, you built your local docker container from that image *before* the breaking change was introduced to chia-blockchain:latest (whatever it was) but I just built my local docker container today, after the breaking change. So my 0.7.2 machinaris is subtlety different than yours because the underlying chia-blockchain:latest branch has been updated in the meantime.

I think 0.7.2 should point to 1.3.5 instead of latest, like the 0.7.1 version does before it (which works great for me today). Pointing to latest seems like a recipe for disaster (if I'm right about my assumptions above)

**Comment by guydavis** [comment]
Would you mind providing the output of:
```
docker exec -it machinaris bash
chia version
git rev-parse --verify HEAD
```
I'm just having trouble seeing what you are reporting.

Also, could you provide a full screenshot of the broken Summary page, along with the [Machinaris logs](https://github.com/guydavis/machinaris/wiki/Troubleshooting)?  That would really help a lot.

**Comment by joshpainter** [comment]
Yep sure, I was on 0.7.1 but here's how I just upgraded to 0.7.2:

- stop container, docker image prune -a to remove image, verify with docker images (only shows hello-world now)
- edit my ~/machinaris/docker-compose.yaml file to go back to "latest" as Machinaris docker generator suggests (I had changed this to v0.7.1 instead as workaround as shown above)
```
version: '3.7' 
services: 
    machinaris: 
        image: ghcr.io/guydavis/machinaris:latest 
```
- docker-compose up -d --force-recreate
- docker ps (to get container id)
- docker exec -it <id> bash
- chia version < shows 1.3.5 (but remember, so does chia-blockchain:latest)
- chia show -s (full node synced)
- chia wallet show (wallet synced)
- chia farm summary (all plots found, sync'd, farming)

However, summary page shows "No blockchains found from any farmers. Just starting up? Please allow at least 15 minutes for blockchains to get started." Workers page shows worker is connected and responding. Wallet page does not show any wallets and shows "No wallet status received. Perhaps just starting? Please allow at least 10 minutes to update. You can also try running "chia wallet show" on your fullnode's in-container shell to verify." Blockchain, keys, connections and pools all load correctly. Main dashboard loads correction and shows "farming" and current stats. But summary and wallet remains empty.

ALSO, I just noticed this error in the log. I think we have our culprit.

    return exp.expect_loop(timeout)
  File "/chia-blockchain/venv/lib/python3.9/site-packages/pexpect/expect.py", line 181, in expect_loop
    return self.timeout(e)
  File "/chia-blockchain/venv/lib/python3.9/site-packages/pexpect/expect.py", line 144, in timeout
    raise exc
pexpect.exceptions.TIMEOUT: Timeout exceeded.
<pexpect.pty_spawn.spawn object at 0x7fc10f9761c0>
command: /chia-blockchain/venv/bin/chia
args: ['/chia-blockchain/venv/bin/chia', 'plotnft', 'show']
buffer (last 100 chars): b"376 (Synced)\r\n2)   1482423266\r\nChoose a wallet key [1-2] ('q' to quit, or Enter to use 4091817376): "
before (last 100 chars): b"376 (Synced)\r\n2)   1482423266\r\nChoose a wallet key [1-2] ('q' to quit, or Enter to use 4091817376): "
after: <class 'pexpect.exceptions.TIMEOUT'>
match: None
match_index: None
exitstatus: None
flag_eof: False
pid: 706
child_fd: 13
closed: False
timeout: 30
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: None
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0.05
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: re.compile(b'Wallet height:.*\r\n')
    1: re.compile(b'Choose wallet key:.*\r\n')
    2: re.compile(b'No online backup file found.*\r\n')

Machinaris seems to be looking for "Choose wallet key" but it was changed to "Choose a wallet key" (notice extra a) with a commit to latest on April 5th  https://github.com/Chia-Network/chia-blockchain/commit/bb57ccffa942afc654a0874bad3bef6613045756 Line 537 and 569

I know 0.7.2 was released after that, so I don't know how it could have passed testing? Unless it was built with an older version of chia-blockchain:latest (before this change) and then "frozen in time" while you worked on 0.7.2 because you didn't pull latest?

Also: perhaps we are talking about different things - does Unraid use a "prebuilt" image that was made back then, perhaps? I'm just following these commands to download and build the image locally today. By editing my docker-compose.yaml file to specify the v0.7.1 tag for your repo (which points to Chia 1.3.2, before the extra a), I'm up and running again.

SO: two suggestions: make a "hotfix" 0.7.3 that points to chia-blockchain 1.3.5 (instead of latest) ASAP. Then update your latest branch to deal with the fun that our friend @paninaro introduced for you. Your latest could still point to chia-blockchain:latest if you want, but seems dangerous for exactly this reason. It seems that your latest branch should always be against a specific Chia branch to avoid this.

Hope that helps! Love Machinaris btw!

**Comment by joshpainter** [comment]
Saw you are already hard at work - hope this PR saves some time (wasn't sure about the `chia_install.sh` stuff but reverted to be sure.

**Comment by joshpainter** [comment]
Also I definitely understand more about docker now and I see what you were saying about the image - I tried to do a docker build on my changes but I'm not sure exactly how that works yet so Imma let you finish lol

**Comment by guydavis** [comment]
Machinaris [version 0.8.0](https://github.com/guydavis/machinaris/releases/tag/0.8.0) has now been released with support for Chia 1.4.0.  Thanks!
